Makefile

```diff
@@ -113,3 +107,71 @@ generate-pip-lockfiles: auth/pinned-requirements.txt
 generate-pip-lockfiles: batch/pinned-requirements.txt
 generate-pip-lockfiles: ci/pinned-requirements.txt
 generate-pip-lockfiles: memory/pinned-requirements.txt
+
+$(HAILTOP_VERSION):
+	$(MAKE) -C hail python/hailtop/hail_version
+
+hail-ubuntu-image: docker/hail-ubuntu
+	$(eval HAIL_UBUNTU_IMAGE := $(DOCKER_PREFIX)/hail-ubuntu:$(TOKEN))
+	python3 ci/jinja2_render.py '{"global":{"docker_prefix":"$(DOCKER_PREFIX)"}}' docker/hail-ubuntu/Dockerfile docker/hail-ubuntu/Dockerfile.out
+	./docker-build.sh docker/hail-ubuntu Dockerfile.out $(HAIL_UBUNTU_IMAGE)
+	echo $(HAIL_UBUNTU_IMAGE) > $@
+
+base-image: hail-ubuntu-image docker/Dockerfile.base
+	$(eval BASE_IMAGE := $(DOCKER_PREFIX)/base:$(TOKEN))
+	python3 ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat hail-ubuntu-image)'"}}' docker/Dockerfile.base docker/Dockerfile.base.out
+	./docker-build.sh . docker/Dockerfile.base.out $(BASE_IMAGE)
+	echo $(BASE_IMAGE) > $@
+
+private-repo-hailgenetics-hail-image: hail-ubuntu-image docker/hailgenetics/hail/Dockerfile $(shell git ls-files hail/src/main hail/python)
+	$(eval PRIVATE_REPO_HAILGENETICS_HAIL_IMAGE := $(DOCKER_PREFIX)/hailgenetics/hail:$(TOKEN))
+	$(MAKE) -C hail wheel
+	cp hail/build/deploy/dist/hail-$$(cat hail/python/hail/hail_pip_version)-py3-none-any.whl .
+	tar -cvf wheel-container.tar hail-$$(cat hail/python/hail/hail_pip_version)-py3-none-any.whl
+	rm hail-$$(cat hail/python/hail/hail_pip_version)-py3-none-any.whl
+	python3 ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat hail-ubuntu-image)'"}}' docker/hailgenetics/hail/Dockerfile docker/hailgenetics/hail/Dockerfile.out
+	./docker-build.sh . docker/hailgenetics/hail/Dockerfile.out $(PRIVATE_REPO_HAILGENETICS_HAIL_IMAGE)
+	rm wheel-container.tar
+	echo $(PRIVATE_REPO_HAILGENETICS_HAIL_IMAGE) > $@
+
+.PHONY: docs
+docs:
+	$(MAKE) -C hail hail-docs-no-test batch-docs
+	gcloud storage cp gs://hail-common/builds/0.1/docs/hail-0.1-docs-5a6778710097.tar.gz .
+	mkdir -p hail/build/www/docs/0.1
+	tar -xvf hail-0.1-docs-5a6778710097.tar.gz -C hail/build/www/docs/0.1 --strip-components 2
+	rm hail-0.1-docs-5a6778710097.tar.gz
+	tar czf docs.tar.gz -C hail/build/www .
+
+website-image: docs
+
+$(SERVICES_IMAGES): %-image: $(SERVICES_IMAGE_DEPS) $(shell git ls-files $$*)
+	$(eval IMAGE := $(DOCKER_PREFIX)/$*:$(TOKEN))
+	python3 ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"'$$(cat hail-ubuntu-image)'"}}' $*/Dockerfile $*/Dockerfile.out
+	./docker-build.sh . $*/Dockerfile.out $(IMAGE)
+	echo $(IMAGE) > $@
+
+ci-utils-image: hail-ubuntu-image $(SERVICES_IMAGE_DEPS) ci/jinja2_render.py ci/wait-for.py ci/create_database.py
```

Comment:
I'm inclined to just use ls-files here for ci again. Seems like something we'd forget to fix.